vignettes/articles/oauth.Rmd

```diff
@@ -56,94 +75,159 @@ client <- oauth_client(
+We can authenticate this request with `req_oauth_auth_code()`, using the same arguments as our previous call to `oauth_flow_auth_code()`:
 
-```{r, eval = FALSE}
-request("https://api.github.com/user") %>% 
-  req_oauth_auth_code(client, auth_url = "https://github.com/login/oauth/authorize") %>% 
+```{r}
+#| eval: false
+req %>% 
+  req_oauth_auth_code(
+    client = github_client(),
+    auth_url = "https://github.com/login/oauth/authorize"
+  ) %>% 
   req_perform() %>% 
-  resp_body_json()
+  resp_body_json() %>% 
+  str()
 ```
 
-This is important because most APIs provide only a short-lived access token that needs to be regularly refreshed using a longer-lived refresh token.
-httr2 will automatically refresh the token if it's expired (i.e. its expiry date is in the past) or if the request errors with a 401 and there's an `invalid_token` error in the `WWW-authenticate` header.
+When you run this code, you'll see something like this, but it will obviously contain information about you, not me.
 
-## Caching
+```{r}
+#| eval: false
+#> List of 32
+#>  $ login        : chr "hadley"
+#>  $ id           : int 4196
+#>  $ node_id      : chr "MDQ6VXNlcjQxOTY="
+#>  $ avatar_url   : chr "https://avatars.githubusercontent.com/u/4196?v=4"
+#>  $ gravatar_id  : chr ""
+#>  $ url          : chr "https://api.github.com/users/hadley"
+#>  $ html_url     : chr "https://github.com/hadley"
+#>  ...
+#>  $ type         : chr "User"
+#>  $ site_admin   : logi FALSE
+#>  $ name         : chr "Hadley Wickham"
+#>  $ company      : chr "@posit-pbc"
+#>  $ blog         : chr "http://hadley.nz"
+#>  $ location     : chr "Houston, TX"
+```
 
-By default, `req_oauth_auth_code()` and friends will cache the token in memory, so that multiple requests in the same session all use the same token.
-In some cases, you may want to save the token so that it's automatically used across sessions.
-This is easy to do (just set `cache_disk = TRUE` in `req_oauth_auth_code()`) but you need to carefully consider the consequences of saving the user's credentials on disk.
+### Token management
 
-httr2 does the best it can to save these credentials securely.
-They are stored in a local cache directory (`oauth_cache_path())` that should only be accessible to the current user, and are encrypted so they will be hard for any package other than httr2 to read.
-However, there's no way to prevent other R code from using httr2 to access them, so if you do choose to cache tokens, you should inform the user and give them the ability to opt-out.
+There are two big reasons to allow httr2 to manage tokens for you.
+The first is that httr2 will automatically refresh the token if it's expired.
+The second is cross-session caching which you can turn on with `cache_disk = TRUE`.
+Turning it on means that your users will only need to re-authenticate once a month, but does means their credentials are saved on disk.
+Reading the "Caching" section in `?req_oauth_auth_code` for more details.
```

Comment:
why not reproduce it here? There are details on the not secret secret at the beginning of the vignette so why not also have the details on the actual secrets here? :smile_cat:

Reply:
Mostly because I don’t want to have the same information in two places because I’ll forgot to update one. Do you think it would make more sense for the vignette to be the primary source and I just link to it from the docs?

Reply:
Yes, it'd be nice for the vignette to have all the info (for Ctrl+F, and for a reader to know they'll have all the info by the time they read it)